### PR TITLE
Also swallow exception when container doesn't exist for DeleteIfExists

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by katmsft — APPROVED** [review]

**Review by katmsft — COMMENTED** [review]
Please add test case for this.

File: sdk/storage/azure-storage-blobs/src/blob_client.cpp [hunk 1]
```diff
@@ -631,7 +631,8 @@ namespace Azure { namespace Storage { namespace Blobs {
     }
     catch (StorageException& e)
     {
-      if (e.StatusCode == Core::Http::HttpStatusCode::NotFound && e.ErrorCode == "BlobNotFound")
+      if (e.StatusCode == Core::Http::HttpStatusCode::NotFound
+          && (e.ErrorCode == "BlobNotFound" || e.ErrorCode == "ContainerNotFound"))
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/src/blob_client.cpp:635]: Unit test?

**Comment by ahsonkhan** [comment]
How did we discover this? What defines what exceptions and error codes we want to swallow?

**Comment by Jinming-Hu** [comment]
> How did we discover this? What defines what exceptions and error codes we want to swallow?

Alignment with .Net

**Pull request merged**